src/modules/ethereum/chains/ChainConfigStore.ts

```diff
@@ -61,10 +85,45 @@ export class ChainConfigStore extends PersistentStore<ChainConfig> {
     const chainStr = chain.toString();
     this.setState((state) =>
       produce(state, (draft) => {
-        remove(draft.ethereumChains, (x) => x.value.chain === chainStr);
+        if (!draft.ethereumChainConfigs) {
+          draft.ethereumChainConfigs = [];
+        }
+        remove(draft.ethereumChainConfigs, (x) => x.id === chainStr);
       })
     );
   }
+
+  private async checkChainsForUpdates() {
+    const { ethereumChainConfigs } = this.getState();
+    if (ethereumChainConfigs?.length) {
+      const updatedEthereumChainConfigs: EthereumChainConfig[] = [];
+      for (const config of ethereumChainConfigs) {
+        if (!isCustomNetworkId(config.id)) {
+          updatedEthereumChainConfigs.push(config);
+          continue;
+        }
+        try {
+          const network = await getNetworkByChainId(config.value.chainId);
+          if (!network) {
+            throw new Error(
+              `Unable to fetch network info by chainId: ${config.value.chainId}`
+            );
+          }
+          updatedEthereumChainConfigs.push({ ...config, id: network.id });
+        } catch {
+          updatedEthereumChainConfigs.push(config);
+        }
+      }
+      this.setState((current) => ({
+        ...current,
+        ethereumChainConfigs: updatedEthereumChainConfigs,
+      }));
+    }
+  }
+
+  async ready(): Promise<void> {
+    return this.isReady ? Promise.resolve() : this.readyPromise;
+  }
```

Comment:
Why overwrite this? Persistent store has the exact same lines

Reply:
I need to check how classes in js works. In some languages rewrite child class's method will not override parent's constructor. And I want new `restore` function to work properly

Reply:
Fixed